packages/ab-testing/src/index.ts

```diff
@@ -25,29 +28,73 @@ function getModuloValue(experiment: string, userId: number | string): number {
     return crc32.calculate(String(userId), crc32.calculate(experiment)) % 100;
 }
 
+function validateCriteria(criteria: ForceInclude, userProfile: UserProfile): boolean {
+    let itemMatchesCriteria = true;
+    for (const key in criteria) {
+        if (!criteria[key].includes(userProfile[key])) {
+            itemMatchesCriteria = false;
+        }
+    }
+
+    return itemMatchesCriteria;
+}
+
+export function validateAllocation(
+    cohort: Cohort,
+    userProfile: UserProfile,
+    userSegmentNum: number
+): boolean {
+    let withinRange = false,
+        fulfillsCriteria = true;
+    if (cohort.allocation) {
```

Comment:
this `if` looks is redundant as we've checked it at line [91](https://github.com/appannie/ab-testing/pull/54/files#diff-fce44da49653c65ce70b06ba4c9f75f5fd04c98bb0f8bd117b0634af2c28f99dR91)